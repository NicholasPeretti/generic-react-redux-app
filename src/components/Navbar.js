import React, { Component } from 'react'
import StyleContext from '../styleContext'
import NavLogo from './NavLogo'
import NavMenu from './NavMenu'
import NavMenuItem from './NavMenuItem'

class Navbar extends Component {
  render () {
    return (
      <StyleContext.Consumer>
        {context => (
          <nav
            style={{
              display: 'flex',
              flexDirection: 'row',
              justifyContent: 'space-between',
              height: '80px',
              borderBottom: `1px solid ${context.colors.gray}`,
              padding: `0px ${context.spacing.padding3}`
            }}
          >
            <NavLogo/>
            <NavMenu>
              <NavMenuItem>Purchase</NavMenuItem>
              <NavMenuItem>My Orders</NavMenuItem>
              <NavMenuItem>Sell</NavMenuItem>
            </NavMenu>
          </nav>
        )}
      </StyleContext.Consumer>
    )
  }
}

export default Navbar
