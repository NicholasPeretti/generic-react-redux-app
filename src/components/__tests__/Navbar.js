import React from 'react'
import { shallow } from 'enzyme'
import PresentationalTest from '../../utils/tests/presentational'
import Navbar from '../Navbar'

describe('Navbar', () => {
  PresentationalTest(Navbar)

  test('Should match snapshot', () => {
    const wrapper = shallow(<Navbar/>)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
