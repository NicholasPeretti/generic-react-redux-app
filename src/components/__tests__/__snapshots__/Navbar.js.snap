// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should match snapshot 1`] = `"<header><nav class=\\"sc-ifAKCX cZmXwS\\"><div class=\\"sc-bdVaJa NFzwl\\"><img src=\\"logo.png\\" alt=\\"Auto1\\" class=\\"sc-bwzfXH foBZsz\\"/></div><ul class=\\"sc-htpNat dUBBzr\\"><li class=\\"sc-bxivhb isUEdk\\">Purchase</li><li class=\\"sc-bxivhb isUEdk\\">My Orders</li><li class=\\"sc-bxivhb isUEdk\\">Sell</li></ul></nav></header>"`;
